src/docfx/build/xref/XrefResolver.cs

```diff
@@ -281,21 +295,33 @@ private void ValidateExternalXref(IReadOnlyDictionary<string, InternalXrefSpec[]
         }
     }
 
-    private static string RemoveSharingHost(string url, string hostName)
+    // legacy hostname and new hostname are treated as the same hostname
+    private string RemoveSharingHost(string url, string hostName)
     {
-        // TODO: this workaround can be removed when all xref related repos migrated to v3
-        if (hostName.Equals("docs.microsoft.com", StringComparison.OrdinalIgnoreCase)
-            && url.StartsWith($"https://review.docs.microsoft.com/", StringComparison.OrdinalIgnoreCase))
+        url = RemoveHostIfMatch(url, hostName);
+
+        if (_hostNameMap.ContainsKey(hostName))
         {
-            return url["https://review.docs.microsoft.com".Length..];
+            // handle the case: hostname is legacy one, but url's hostname is new one
+            url = RemoveHostIfMatch(url, _hostNameMap[hostName]);
         }
-
-        if (url.StartsWith($"https://{hostName}/", StringComparison.OrdinalIgnoreCase))
+        if (_reversedHostNameMap.ContainsKey(hostName))
```

Comment:
Why do we need to remove again as line 301 and line 303 already remove both host names?

Reply:
For handle the cases: 
1. hostname is legacy one, but url's hostname is new one
2. hostname is new one, but url's hostname is legacy one

Reply:
What will be the mapping like if host name is the new one? I suppose the mapping should always be: legacy host name -> new host name - Am I understanding correctly?